infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/namespace/KubernetesObjectUtil.java

```diff
@@ -95,12 +96,26 @@ public static void putSelector(Service target, String key, String value) {
    * quantity.
    */
   public static PersistentVolumeClaim newPVC(String name, String accessMode, String quantity) {
-    return new PersistentVolumeClaimBuilder()
-        .withNewMetadata()
-        .withName(name)
-        .endMetadata()
-        .withNewSpec()
-        .withAccessModes(accessMode)
+    return newPVC(name, accessMode, quantity, null);
+  }
+
+  /**
+   * Returns new instance of {@link PersistentVolumeClaim} with specified name, accessMode, quantity
+   * and storageClassName.
+   */
+  public static PersistentVolumeClaim newPVC(
```

Comment:
What do you think about rewriting this method in the following way?
```java
  public static PersistentVolumeClaim newPVC(
      String name, String accessMode, String quantity, String storageClassName) {
    return new PersistentVolumeClaimBuilder()
        .withNewMetadata()
        .withName(name)
        .endMetadata()
        .withNewSpec()
        .withAccessModes(accessMode)
        .withStorageClassName(isNullOrEmpty(storageClassName) ? null : storageClassName)
        .withNewResources()
        .withRequests(ImmutableMap.of(STORAGE_PARAM, new Quantity(quantity)))
        .endResources()
        .endSpec()
        .build();
  }
```

Reply:
I've not tested but I think your suggestion won't work fine.

Referring to https://kubernetes.io/blog/2017/03/dynamic-provisioning-and-storage-classes-kubernetes/, 

> * If storageClassName is not specified in the PVC, the default storage class will be used for provisioning.
>    * Existing, “Available”, PVs that do not have the default storage class label will not be considered for binding to the PVC
> * If storageClassName is set to an empty string (‘’) in the PVC, no storage class will be used (i.e.; dynamic provisioning is disabled for this PVC)
>    * Existing, “Available”, PVs (that do not have a specified storageClassName) will be considered for binding to the PVC

The current behavior of Che is "dnyamic provisioning"
So we must not call `withStorageClassName()` if `che.infra.kubernetes.pvc.storage_class_name` is empty.

(My choice causes the another issue that no one can disable dynamic provisioning. But it isn't serious in usual use cases in Che.)